### PR TITLE
SEN-1322 -- S3 listObjects as a monadic stream.

### PR DESCRIPTION
**Pull request by nackjicholson**
Had to write this one method when I was doing the downtime recovery script to pull files from s3 which had last modified times between a certain range.

I wanted this code in shareable spot, and I've also done some similar work with `sqs.receiveMessage` in our digest-chunker, so I thought I'd make `aws-most` it's own project. Also seems to mesh really well with my effort to promisificate all the aws clients in `aws-promised`.

**To Test**

Review code
Run the unit tests, verify coverage. `npm test`

Bonus points:

Make a testing bucket with some files, alter `examples/s3-listObjects.js` to make sense for the files in your bucket and run it.

test the pagination ability by putting more than a thousand files in a bucket and then running the listObjects stream.

### CODE REVIEW

**Comment by nackjicholson** [comment]
`npm test`

**Comment by nackjicholson** [comment]
My use case of copying files that fall in a certain time range has surfaced a problem when casting a wide net for that time range. A major one in fact. I hit an `EMFILE` error. [stackoverflow](http://stackoverflow.com/questions/8965606/node-and-error-emfile-too-many-open-files)

By making this a stream I've essentially removed the safety net that a 1000 file paginated request provides, in favor of an unstoppable firehose of file information.

If we were going to mostify unfolded requests like this we would need a way to pause them. Not quite sure how I would do that yet -- but off the top of my head `s3Most would probably have to recieve a message to "stop"/"pause" after it's current `unfold`, and save the last Key it retrieved for later use if the stream needs to be started back up.

I may close this PR, just don't see it really being useful until I spike on it more. This is a good confirmation that streaming all of the AWS things is going to be harder than I thought though :cry:

**Comment by nackjicholson** [comment]
:ok: Made some changes. Works much better.

File: test/s3/listObjects/unfoldObjects.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+      s3.then.calledOnce.should.equal(true);
+      s3.then.args[0].should.have.length(1);
+      s3.then.args[0][0].should.equal(responseHandlerFn);
+
+      result.should.eql('test.promise');
+    }
+  );
+
+  //describe('handl', function() {
```
Comment (nackjicholson) [inline test/s3/listObjects/unfoldObjects.js]: Uhhhhhhhm...

**Comment by cberube** [comment]
:ok: Works as described -- event tried the bonus steps.

**Pull request merged**